sstable/block.go

```diff
@@ -1008,7 +1008,27 @@ func (i *blockIter) SeekLT(key []byte, flags base.SeekLTFlags) (*InternalKey, ba
 		i.offset = decodeRestart(i.data[i.restarts+4*(index-1):])
 		if index < i.numRestarts {
 			targetOffset = decodeRestart(i.data[i.restarts+4*(index):])
+
```

Comment:
If you like this approach better, I'll add a big comment explaining what's going on here.